internal/controller/nonadminrestore_controller.go

```diff
@@ -333,6 +333,28 @@ func (r *NonAdminRestoreReconciler) createVeleroRestore(ctx context.Context, log
 	}
 
 	if veleroRestore == nil {
```

Comment:
is this approach ok? when we see NAR associated velero restore got deleted, we just inform this in status to non admin user

We could also just delete it. I am afraid this could lead to bad UX, where something by mistake deletes velero restore and nona mdin user does not know why its restore was deleted (but that is only a CR, restore operation would not be undone)

Reply:
What would happen if we re-create Velero Restore? So NAR controller reacting on the Velero Restore objects which has NAC labels and ensures the reconcile of NAC re-creates them when they got deleted?

Reply:
currently, if related velero restore is deleted, it would get recreated by controller

with this PR, if related velero restore is deleted, and then somehow recreated (with same labels and annotations) NAB status would update to match new velero restore status

Reply:
I think status update on NAB should be fine